source/module_esolver/esolver_lj.cpp

```diff
@@ -14,14 +14,122 @@ namespace ModuleESolver
         lj_force.create(ucell.nat, 3);
         lj_virial.create(3, 3);
 
-        lj_rcut = inp.mdp.lj_rcut;
-        lj_epsilon = inp.mdp.lj_epsilon;
-        lj_sigma = inp.mdp.lj_sigma;
-
-        GlobalV::SEARCH_RADIUS = (lj_rcut + 2.0) * ModuleBase::ANGSTROM_AU;
-        lj_rcut *= ModuleBase::ANGSTROM_AU;
-        lj_epsilon /= ModuleBase::Ry_to_eV;
-        lj_sigma *= ModuleBase::ANGSTROM_AU;
+        lj_rule = inp.mdp.lj_rule;
+        int ntype = ucell.ntype;
+        lj_rcut.create(ntype, ntype);
+        lj_c6.create(ntype, ntype);
+        lj_c12.create(ntype, ntype);
```

Comment:
I am a little bit worried here partially because I can hardly remember the behavior of default destructor of class ModuleBase::matrix, will it always release memory automatically?

Reply:
Yes, the default destructor of ModuleBase::matrix will always release memory automatically.